modules/45_carbonprice/functionalForm/postsolve.gms

```diff
@@ -6,92 +6,233 @@
 *** |  Contact: remind@pik-potsdam.de
 *** SOF ./modules/45_carbonprice/functionalForm/postsolve.gms
 
-*** Only run postsolve if cm_iterative_target_adj is not equal to 0.
-if((cm_iterative_target_adj ne 0),
+***-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+*** Part 0 (Actual CO2 budget): If iterative_target_adj = 0, 7 or 9, compute actual CO2 peak budget in current iteration. If iterative_target_adj = 5, compute actual CO2 end-of-century budget in current iteration. 
```

Comment:
Instead of 3 different budget computations (for each cm_iterative_target_adj), compute `p45_actualbudgetco2(ttot)` once, and than compute `s45_actualbudgetco2` as peak budget or end-of-century budget depending on cm_iterative_target_adj. As knowing the budget is also useful when debugging/analyzing runs with exogenous carbon prices, the computation is also done if cm_iterative_target_adj = 0.